typespec-tests/tsp/arm.tsp

```diff
@@ -139,7 +177,16 @@ interface CustomTemplateResourceInterface
         IfMatchParameters<CustomTemplateResource> &
         IfNoneMatchParameters<CustomTemplateResource>
     > {
+  @sharedRoute
+  @added(Versions.v2022_09_01_preview)
+  @removed(Versions.v2023_11_01)
   update is ArmResourceUpdateSync<CustomTemplateResource>;
+
+  #suppress "@azure-tools/typespec-azure-core/no-operation-id" "Use operationId to keep same name as deprecated update operation"
+  @sharedRoute
+  @added(Versions.v2023_11_01)
+  @operationId("CustomTemplateResourceInterfaces_Update")
```

Comment:
example: https://github.com/Azure/azure-rest-api-specs/blob/d53388b8b7574dfe861a9e6863bd4d137168cbe5/specification/containerservice/Fleet.Management/fleetmember.tsp#L100

This is not supported yet. Will fix in following PR.

Reply:
I don't know whether we should honor `@operationId`. Likely not, for greenfield.